### ISSUE TITLE
`button-has-type` marks too much of the JSX expression

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno_lint#1391]
### Lint Name

`button-has-type`

### Code Snippet

```tsx
return (
  <button
    autofocus
  >
    Foo bar
  </button>
);
```

### Expected Result

Only the `button` identifier is highlighted.

### Actual Result

The entire open part of the JSX expression for `button` is marked, which can lead to very long lint diagnostics.

### Version

2.1.7+e475749

**Comment by lucacasonato** [comment]
Same applies to `no-danger` and `jsx-no-danger-with-children`.

**Comment by lucacasonato** [comment]
And also `jsx-key`

**Comment by marvinhagemeister** [comment]
That seems to be a general issue with the linter. The nodes passed to the reporting functions are correct. I'll look into this.

**Comment by marvinhagemeister** [comment]
@lucacasonato not able to reproduce with `jsx-key`. It reports the correct range for me:

```sh
 --> /Users/marvinh/dev/test/deno-lint-range/foo.tsx:2:10
  | 
2 |   return <div>foo</div>;
  |          ^^^^^
  = hint: Add a 'key' prop

  docs: https://docs.deno.com/lint/rules/jsx-key
```

**Comment by lucacasonato** [comment]
@marvinhagemeister add some attributes to the div / make the opening tag multi line. It will highlight the entire opening tag rather than just the identifier
